### PR TITLE
[INFREQ-643] Warn on RunTask failures

### PR DESCRIPTION
**Pull request by liath**
Might want to also log, I'll have to devise a way to cause the API to fail to test this

### CODE REVIEW

**Review by pdasilva — APPROVED** [review]

**Review by ohookins — APPROVED** [review]

**Comment by liath** [comment]
Added unittests across index, not terribly proud of them though

File: index.js [hunk 1]
```diff
@@ -74,6 +74,12 @@ module.exports = function (options, cb) {
       return taskRunner.runPromisified(params);
     })
     .then((taskDefinition) => {
+      if (taskDefinition.failures) {
+        console.error("Task failed to launch:")
```
Comment (ohookins) [inline index.js:78]: I think it's fine to just throw the error - anything interested should catch it and log as appropriate. This is intended to be used as a library after all.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "ecs-task-runner": "./bin/ecs-task-runner"
   },
   "scripts": {
-    "test": "NODE_ENV=test ./node_modules/.bin/mocha --recursive"
+    "test": "NODE_ENV=test ./node_modules/.bin/mocha --parallel --recursive"
```
Comment (ohookins) [inline package.json:13]: would avoid parallel usage, I remember some problems with one of these codebases where it was not really concurrency safe

File: test/index.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    ecsMock.on(RunTaskCommand).callsFake(async params => {
+      expect(params.taskDefinition).to.eql(options.taskDefinitionArn)
+
+      return {
+        failures: [{ reason }]
+      }
+    });
+
+    try {
+      await index(options);
```
Comment (ohookins) [inline test/index.js:105]: why not something like `expect(index(options).to.throw ...` ?

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    // thread the randon EOF through to kill the stream
+    let eofSet;
+    const eof = new Promise(r => {
+      eofSet = r;
+    })
+    ecsMock.on(RunTaskCommand).callsFake(async params => {
+      expect(params.taskDefinition).to.eql(options.taskDefinitionArn)
+
+      eofSet(params.overrides.containerOverrides[0].command[2].split(' ')[9])
```
Comment (ohookins) [inline test/index.js:49]: I don't really get what this does

File: package.json (hunk 2 above)
Comment (liath) [reply package.json:13]: Well, this is one of them apparently. I removed parallel and now tests won't pass lol

File: index.js (hunk 1 above)
Comment (liath) [reply index.js:78]: Will do

File: package.json (hunk 2 above)
Comment (liath) [reply package.json:13]: https://github.com/m-radzikowski/aws-sdk-client-mock/issues/64
Apparently mocha has weird behavior with aws-sdk-client-mock, shuffling up the mockClient invocations resolves it.

File: test/index.js (hunk 3 above)
Comment (liath) [reply test/index.js:105]: It didn't work ¯\_(ツ)_/¯
expect.js hasn't been updated in a decade, maybe they don't support promises?

Comment (liath) [reply test/index.js:105]: I think mocha and expect.js might be kind dead projects. Something that annoyed my the whole time in working on this is that errors were always "test timed out" and apparently that's just mocha being terrible with promises.

File: test/index.js (hunk 4 above)
Comment (liath) [reply test/index.js:49]: I'll add a comment, but yeah this is ugliest bit of this. This is the only place we can grab the random value the log stuff uses to detect the end of the task run.

**Pull request merged**